src/agents/policy_retriever_agent.py

```diff
@@ -0,0 +1,135 @@
+import os
+import sys
+sys.path.append("..")
+
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_community.vectorstores import Chroma
+from openai import OpenAI, AzureOpenAI
+
+import config as cfg
+from helpers.embedding_helpers import OpenAIEmbeddingFunction
+
+# Environment setup
+os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'
+
+# Fuse to blob storage to access chroma db and policy documents
+os.system('sudo blobfuse /home/azureuser/cloudfiles/code/blobfuse/rndaistoragemeldingen --tmp-path=/mnt/resource/blobfusetmp_meldingen --config-file=/home/azureuser/cloudfiles/code/blobfuse/fuse_connection_meldingen.cfg -o attr_timeout=3600 -o entry_timeout=3600 -o negative_timeout=3600 -o allow_other -o nonempty')
+
+class PolicyRetrieverAgent:
+
+    def __init__(self, melding):
+        self.melding = melding
+
+    def retrieve_policy(self):
+        summarized_melding = self.summarize_melding()
+        context_text, sources = self.search_db(summarized_melding)
+        prompt = self.create_prompt(context_text)
+        response_text = self.invoke_model(prompt)
+        formatted_response = self.format_response(response_text, sources)
+        return formatted_response
+
+    def summarize_melding(self):
+        summarize_prompt = ChatPromptTemplate.from_template(cfg.SUMMARIZE_MELDING_TEMPLATE)
+        prompt = summarize_prompt.format(
+            melding=self.melding,
+        )
+        summarized_query = self.invoke_model(prompt, summarize=True)
+        
+        return summarized_query
+
+    def search_db(self, summarized_melding):
+        """
+        Search the Chroma database using the summarized melding,
+        combining and deduplicating results.
+
+        Args:
+            summarized_melding (str): The summarized melding text.
+
+        Returns:
+            tuple: A tuple containing:
+                - context_text (str): Combined context text from the search results.
+                - sources (list): List of sources used in the search.
+        """
+        db = Chroma(persist_directory=cfg.CHROMA_PATH, embedding_function=OpenAIEmbeddingFunction())
```

Comment:
Why is this being initialized at every search? Isn't it better to initialize the DB one time (when initializing a general retriever tool) and then just search every time a new melding comes in?

Reply:
Currently, this setup simplifies testing by allowing us to run only the central_agentic_agent.py file without launching the entire frontend. If the database were initialized as part of the full chatbot session, testing the central agent (including the policy retriever tool) would require running the frontend too, which would be less efficient.

Reply:
But this initialization doesn't have anything to do with the front end, right? It's a matter of moving it out of here as a class attribute (and then also further get the tool instantiation out of the "tool function", but again as a class attribute of the central agent (that is the setup of the meldingen retrieval tool btw, you can see it in that PR)

Reply:
Ah yes I get what you mean. I'll make that change and initialize it as part of the object (also with regard to the tool function)